src/services/behaviors/index.js

```diff
@@ -88,3 +90,50 @@ export const setIndicatorsAfterLoad = (
   newRoot = toggleIndicators(hideIndicatorIds, true, newRoot);
   return newRoot;
 };
+
+/**
+ * Returns a new Document object where the given action was applied to the target element
+ * with the new element.
+ */
+export const performUpdate = (
+  action: UpdateAction,
+  targetElement: Element,
+  newElement: Element,
+): Document => {
+  if (action === 'replace') {
```

Comment:
we have a dictionary `ACTIONS` in `src/types.js` which might be good to use in place of hardcoded action values